app/uk/gov/hmrc/gform/sharedmodel/TaxPeriods.scala

```diff
@@ -49,3 +50,41 @@ case class TaxResponse(id: HmrcTaxPeriod, obligation: Obligation)
 object TaxResponse {
   implicit val format: OFormat[TaxResponse] = Json.format[TaxResponse]
 }
+
+case class TaxPeriodIdentifier(idType: IdType, idNumber: IdNumber, regimeType: RegimeType)
+
+object TaxPeriodIdentifier {
+  implicit val format: OFormat[TaxPeriodIdentifier] = Json.format[TaxPeriodIdentifier]
+}
+
+case class AllInfo(
```

Comment:
This name needs to be better.  Perhaps something along the lines of TaxPeriodMapping